File: stack in array.cpp
#include <iostream>
using namespace std;
int main() {
    int stack[100],x,i=0,j,top,choice=1;
    while(1){
        cout <<"\n1.push, 2.pop, 3.top, 4.print 0.exit\n";
        cin >>choice;
        switch(choice){
            case 1: cout << "Push:";
                    cin >> x;
                    stack[i]=x;
                    top=stack[i];
                    i++;
                    break;
            case 2: if(i>=0){
                        if(i>0)
                            cout <<"Pop:"<<top;
                        i--;
                        if(i>0)
                            top=stack[i-1];
                    }
                    else
                        cout <<"stack is empty\n";
                    break;
            case 3: if(i>=0)
                        cout<<"top:\n"<<top;
                    else
                        cout<<"stack is empty\n";
                    break;
            case 4: cout<<"stack:";
                    if(i>=0){
                        for(j=0;j<i;j++)
                           cout<<stack[j]<<",";
                    }
                    else
                        cout<<"stack is empty\n";
                    break;
            case 0: exit(1);
            default:cout<<"Error! choice is not correct\n";
        }
    }
	return 0;
}

File: stack in array.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);  
        int[] stack = new int[100];
        int x,i=0,j,top=0,choice=1;
        while(true){
            System.out.println("\n1.push, 2.pop, 3.top, 4.print 0.exit\n");
            choice= sc.nextInt();
            switch(choice){
                case 1: System.out.println("Push:");;
                        x = sc.nextInt();  
                        stack[i]=x;
                        top=stack[i];
                        i++;
                        break;
                case 2: if(i>=0){
                            if(i>0)
                                System.out.println("Pop:"+top);
                            i--;
                            if(i>0)
                                top=stack[i-1];
                        }
                        else
                            System.out.println("stack is empty\n");
                        break;
                case 3: if(i>=0)
                            System.out.println("Pop:"+top);
                        else
                            System.out.println("stack is empty\n");
                        break;
                case 4: System.out.println("stack:");
                        if(i>=0){
                            for(j=0;j<i;j++)
                                System.out.println(stack[j]+",");
                        }
                        else
                            System.out.println("stack is empty\n");
                        break;
                    
                case 0: System.exit(0);
                default:System.out.println("Error! choice is not correct\n");
            }
        }
    }
}